src/core/logical_layers/renderers/BivariateRenderer.tsx

```diff
@@ -211,9 +210,7 @@ export class BivariateRenderer extends LogicalLayerDefaultRenderer {
 
       if (!fillColor || isFillColorEmpty(fillColor) || !feature.properties) return true;
 
-      const showValues = userResourceAtom
-        .getState()
-        .data?.hasFeature(AppFeature.BIVARIATE_MANAGER);
+      const showValues = featureFlagsAtom.getState()[FeatureFlag.BIVARIATE_MANAGER];
```

Comment:
isn't calling `.getState()` in every `_updateMap` method potentially expensive perfomance wise?
also bivariate matrix that is pretty slow by itself relies on this file too

Reply:
it's inside clickHandler, so it will be triggered only if you click on a hexagon